packages/manager/modules/pci/src/projects/project/storages/snapshots/snapshots.service.js

```diff
@@ -0,0 +1,137 @@
+import map from 'lodash/map';
+import find from 'lodash/find';
+import uniq from 'lodash/uniq';
+import reduce from 'lodash/reduce';
+import VolumeSnapshot from './snapshot.class';
+
+export default class PciProjectStorageSnapshotsService {
+  /* @ngInject */
+  constructor(
+    $q,
+    OvhApiCloudProject,
+    OvhApiCloudProjectVolumeSnapshot,
+  ) {
+    this.$q = $q;
+    this.OvhApiCloudProject = OvhApiCloudProject;
+    this.OvhApiCloudProjectVolumeSnapshot = OvhApiCloudProjectVolumeSnapshot;
+  }
+
+  getAll(projectId) {
+    let snapshots;
+
+    return this.OvhApiCloudProjectVolumeSnapshot
+      .v6()
+      .query({
+        serviceName: projectId,
+      })
+      .$promise
+      .then((results) => {
+        snapshots = [...results];
+        const volumeIds = uniq(
+          reduce(
```

Comment:
Could it make sense to map instead ? 
```suggestion
         map(snapshots, 'volumeId')
```